### PR TITLE
DOC: Added note for `corr` 

### LINKED ISSUE #49939: DOC: better explain the automatic alignment process
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

Throughtout https://pandas.pydata.org/pandas-docs/stable/index.html

### Documentation problem

> I'd like to mention documentation again, for emphasis.  A someone who found the automatic data alignment 'confusing', I think I offer a unique perspective to this problem (at least on this thread).  For example, I'm currently dealing with numpy shape.  When looking at the documentation for np.shape it would have been helpful if it simply mentioned that it was row X columns when I looked np.shape rather the just "array dimensions" (same amount of characters!).  Smirk, if you wish, but these minor details here and there repeated can go along way to helping folks new to these frameworks out.
> 

Originally reported by @blazespinnaker [here](https://github.com/pandas-dev/pandas/pull/49694#issuecomment-1328701631)

### Suggested fix for documentation

> For this problem, a quick note in different operations such as sum could say 
> 
> `
> *Note automatic data alignment:  as with all pandas operations, automatic data alignment is performed.  If sum does not find values with matching indicies than NaN will used as the total.   
> `
> 
> An example would even be even more awesome.
> 
> 
> Note that my text is probably very poor, but ideally it would be written in a way to make alignment less confusing to new users.
> 
> 

Originally reported by @blazespinnaker [here](https://github.com/pandas-dev/pandas/pull/49694#issuecomment-1328701631)

### PR DESCRIPTION
**Pull request by rsm-23**
- [ ] closes #49939 
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

**Comment by rsm-23** [comment]
@MarcoGorelli please review this when possible.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks for your PR

sorry, I didn't explain the issue very clearly

the original issue was about `Series.corr`, so let's add an example there instead

How about an example in which we show:
```python
>>> s1 = pd.Series([1, 2, 3], index=[0, 1, 2])
>>> s2 = pd.Series([1, 2, 3], index=[2, 1, 0])
```
and note that `s1.corr(s2)` gives `-1.0` because pandas aligns the indexes?

**Review by MarcoGorelli — APPROVED** [review]
wonderful, thanks @rsm-23 !

let's close the issue with this one, if other places come up where results are unexpected we can document more clearly there too, otherwise the issue will stay open forever

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
the changes look good, but it's failing the doc build

```
/home/runner/work/pandas/pandas/pandas/core/groupby/generic.py:docstring of pandas.core.groupby.generic.SeriesGroupBy.corr:33: ERROR: Content block expected for the "note" directive; none found.
/home/runner/work/pandas/pandas/pandas/core/groupby/generic.py:docstring of pandas.core.groupby.generic.SeriesGroupBy.corr:34: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/runner/work/pandas/pandas/pandas/core/series.py:docstring of pandas.core.series.Series.corr:33: ERROR: Content block expected for the "note" directive; none found.
/home/runner/work/pandas/pandas/pandas/core/series.py:docstring of pandas.core.series.Series.corr:34: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

could you build this page locally please and make sure it passes? then it should be good

```
cd doc
python make.py --single pandas.Series.corr
```

**Comment by rsm-23** [comment]
Sure @MarcoGorelli

**Comment by rsm-23** [comment]
I tried building this page using 
`python make.py --single pandas.Series.corr`

**Comment by MarcoGorelli** [comment]
don't know, can you check the contributing guide for documentation please?

File: pandas/core/series.py [hunk 1]
```diff
@@ -2769,6 +2769,11 @@ def corr(
         float
             Correlation with other.
 
+        .. note::
+
+            Automatic data alignment: as with all pandas operations, automatic data alignment is performed for this method.
+            ``corr`` automatically considers values with matching indices.
```
Comment (MarcoGorelli) [inline pandas/core/series.py:2775]: maybe just add this to the "notes" section below?

Comment (rsm-23) [reply pandas/core/series.py:2775]: okay, sure

**Comment by rsm-23** [comment]
@MarcoGorelli can you please check the docs?

**Review by MarcoGorelli — APPROVED** [review]
thanks @rsm-23 !

**Pull request merged**